src/graph/service/QueryInstance.cpp

```diff
@@ -93,7 +93,8 @@ Status QueryInstance::validateAndOptimize() {
   // Optimize the query, and get the execution plan. We should not pass the optimizer errors to user
   // since the message is often not easy to understand. Logging them is enough.
   if (auto status = findBestPlan(); !status.ok()) {
-    LOG(ERROR) << "Error found in optimization stage: " << status.message();
+    LOG(ERROR) << "Error found in optimization stage for query: " << rctx->query()
+               << ", error: " << status.message();
     return Status::Error(
         "There are some errors found in optimizer, "
         "please contact to the admin to learn more details");
```

Comment:
The error message on line 100 is not appropriate, it does not prevent the error message from being passed to the user, and it is more confusing than the original error message.

Reply:
Actually, I am not sure what's the better way to handle the optimizer errors. I need to open an issue about this and research how to do that in other db.

Reply:
#5033